voyager/plugins/zkgm-filter/src/main.rs

```diff
@@ -468,18 +469,53 @@ impl Module {
     }
 }
 
+/// NOTE: Assumes the tx contains only one MsgExecuteContract.
 fn valid_checksum_cosmos(tx_bytes: &[u8]) -> bool {
     let tx_raw = TxRaw::decode_as::<Proto>(tx_bytes).expect("invalid transaction?");
-    let mut tx_body = <TxBody<Any<MsgExecuteContract>>>::decode_as::<Proto>(&tx_raw.body_bytes)
-        .expect("invalid auth info?");
 
-    let msg = tx_body
-        .messages
-        .pop()
-        .expect("must contain at least one message");
+    let tx_body = match <TxBody<RawAny>>::decode_as::<Proto>(&tx_raw.body_bytes) {
```

Comment:
why not let-else?

Reply:
i want the Err value on error so i can print it